### ISSUE TITLE
Properly verify message id existence and spend in the block executor

### ISSUE THREAD

**Issue by Voxelot** [FuelLabs/fuel-core#512]
When utxo_validation is enabled, the block execution logic should verify the existence of bridge messages. The executor also needs to mark the message as spent to prevent reuse.

This work was deferred from the main task (#510) of upgrading to the latest versions of fuel-tx and fuel-vm, ~~and is blocked on #485~~ as well to check if the message exists in the database.

Test Cases:

- Block execution succeeds when a tx includes a message with message da_height <= block da_height
- Block execution fails if tx spends a message id that does not exist in db
- Block execution fails if tx spends a message id with da_height > block da_height
- Block execution fails if tx spends a message id that has already been spent
